src/index.js

```diff
@@ -949,28 +1064,39 @@ export default function (babel) {
             " with a (key, value) pair.");
           }
 
+          const node = tailPath.node;
+          const n = (...args) => nodeAt(node, ...args);
           const keyExpr = seqExpr.expressions[0];
           const valExpr = seqExpr.expressions[1];
 
-          return t.assignmentExpression(
+          return n("assignmentExpression",
             "=",
-            t.memberExpression(id, keyExpr, true),
+            n("memberExpression", id, keyExpr, true),
             valExpr
           );
         });
 
-        path.replaceWith(wrapComprehensionInIife(id, t.objectExpression([]), path.node.loop));
+        path.replaceWith(
+          wrapComprehensionInIife(
+            id,
+            nodeAt(path.node, "objectExpression", []),
+            path.node.loop
+          )
+        );
       },
 
       TildeCallExpression: {
         // run on exit instead of enter so that SafeMemberExpression
         // can process differently from a wrapping CallExpression
         // eg; `a?.b~c()` -> `a == null ? null : c(a.b)`
         exit(path) {
-          const callExpr = t.callExpression(path.node.right, [
-            path.node.left,
-            ...path.node.arguments,
-          ]);
+          // XXX: this might be breaking source maps because it makes impossible
+          // ordering, right appears after left in original source but before
```

Comment:
The general rule should be, I think, that the sourcemap points to the source that generated the error or, if that source does not appear in the original lsc (because it was generated by a transform) then to the nearest position where we are inserting it. That is the basic pattern I'm using.